src/export.rs

```diff
@@ -0,0 +1,235 @@
+            hands_size: [state.hands[0].len(), state.hands[1].len()],
+            in_play_pokemon: [
+                [
+                    state.in_play_pokemon[0][0].as_ref().map(PokemonState::from),
+                    state.in_play_pokemon[0][1].as_ref().map(PokemonState::from),
+                    state.in_play_pokemon[0][2].as_ref().map(PokemonState::from),
+                    state.in_play_pokemon[0][3].as_ref().map(PokemonState::from),
+                ],
+                [
+                    state.in_play_pokemon[1][0].as_ref().map(PokemonState::from),
+                    state.in_play_pokemon[1][1].as_ref().map(PokemonState::from),
+                    state.in_play_pokemon[1][2].as_ref().map(PokemonState::from),
+                    state.in_play_pokemon[1][3].as_ref().map(PokemonState::from),
+                ],
+            ],
+            current_energy: state.current_energy.map(|e| format!("{:?}", e)),
+            has_played_support: state.has_played_support,
+            has_retreated: state.has_retreated,
+        }
+    }
+}
+
+impl From<&PlayedCard> for PokemonState {
+    fn from(card: &PlayedCard) -> Self {
+        PokemonState {
+            card_name: card.card.get_name().to_string(),
+            hp: card.total_hp,
+            remaining_hp: card.remaining_hp,
+            attached_energy: card.attached_energy.iter().map(|e| format!("{:?}", e)).collect(),
+            attached_tool: card.attached_tool.map(|t| format!("{:?}", t)),
+            played_this_turn: card.played_this_turn,
+            poisoned: card.poisoned,
+            paralyzed: card.paralyzed,
+            asleep: card.asleep,
+            stage: match &card.card {
+                crate::types::Card::Pokemon(pokemon) => pokemon.stage,
+                _ => 0, // Shouldn't happen for played cards
+            },
+        }
+    }
+}
+
+impl From<&Action> for ActionData {
+    fn from(action: &Action) -> Self {
+        ActionData {
+            action_type: format!("{:?}", std::mem::discriminant(&action.action)),
+            description: format!("{}", action.action),
+            actor: action.actor,
+        }
+    }
+}
+
+impl Default for TrainingData {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// Event handler for collecting training data
+pub struct TrainingDataCollector {
+    pub training_data: TrainingData,
+    current_game: Option<GameData>,
+}
+
+impl TrainingDataCollector {
+    pub fn new() -> Self {
+        Self {
+            training_data: TrainingData::new(),
+            current_game: None,
+        }
+    }
+
+    pub fn save_to_file(&self, path: &str) -> Result<(), Box<dyn std::error::Error>> {
+        self.training_data.save_to_file(path)
+    }
+}
+
+impl Default for TrainingDataCollector {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+use crate::simulation_event_handler::SimulationEventHandler;
+
+impl SimulationEventHandler for TrainingDataCollector {
```

Comment:
Great call on using the SimulationEventHandler for this. I think doing the above `impl SimulationEventHandler for TrainingDataCollector` is the right approach.